packages/core/src/odata/v4/payload-value-converter.ts

```diff
@@ -40,117 +43,74 @@ type EdmTypeMapping = {
-  if (!timeComponents) {
-    throw new Error(`Failed to parse the value: ${value} to time.`);
+function edmDateToMoment(date: string): Moment {
+  const parsed = moment.utc(date, 'YYYY-MM-DD', true);
+  if (!parsed.isValid()) {
+    throw new Error(
+      `Provided date value ${date} does not follow the Edm.Date pattern: YYYY-MM-DD`
+    );
   }
-  return {
-    hours: parseInt(timeComponents[1], 10),
-    minutes: parseInt(timeComponents[2], 10),
-    seconds: parseInt(timeComponents[3], 10)
-  };
-};
-
-const fromBigNumber = (value: BigNumber): string =>
-  (value as BigNumber).toString();
-const fromTime = (value: Time): string =>
-  'PT' +
-  leftpad(value.hours, 2) +
-  'H' +
-  leftpad(value.minutes, 2) +
-  'M' +
-  leftpad(value.seconds, 2) +
-  'S';
-
-/**
- * @hidden
- * This function can be used for both Edm.DateTime and and Edm.DateTimeOffset.
- */
-export function edmDateTimeToMoment(edmDateTime: string): Moment {
-  const dateTimeOffsetComponents = /.*\((-?\d+)(?:([\+-])(\d{4})\))?/.exec(
-    edmDateTime
-  );
-  if (!dateTimeOffsetComponents) {
-    throw new Error(`Failed to parse edmDateTime: ${edmDateTime} to moment.`);
-  }
-
-  const timestamp = moment(parseInt(dateTimeOffsetComponents[1]));
-
-  if (dateTimeOffsetComponents[2] && dateTimeOffsetComponents[3]) {
-    const offsetMultiplier = dateTimeOffsetComponents[2] === '+' ? 1 : -1;
-    const offsetInMinutes = parseInt(dateTimeOffsetComponents[3]);
-
-    return timestamp.utc().utcOffset(offsetMultiplier * offsetInMinutes);
-  }
-
-  return timestamp;
+  return parsed;
 }
 
-/**
- * @hidden
- * This function can be used for both Edm.DateTime and and Edm.DateTimeOffset.
- */
-export function momentToEdmDateTime(momentInstance: Moment): string {
-  const timestamp = momentInstance.unix() * 1000;
-
-  // For some reason isUtc() returns wrong values here, so we use the internal flag directly
-  if (momentInstance['_isUTC']) {
-    const offset = Math.abs(momentInstance.utcOffset());
-    const operator = momentInstance.utcOffset() >= 0 ? '+' : '-';
-    return `/Date(${timestamp}${operator}${leftpad(offset, 4)})/`;
+function edmDateTimeOffsetToMoment(dateTime: string): Moment {
+  const prefix = 'YYYY-MM-DDTHH:mm';
+  // In moment the Z is either Offset from UTC as +-HH:mm, +-HHmm, or Z
+  const validFormats = [`${prefix}Z`, `${prefix}:ssZ`, `${prefix}:ss.SSSZ`];
+  const parsed = moment(dateTime, validFormats, true);
+  if (!parsed.isValid()) {
+    throw new Error(
+      `Provided date-time value ${dateTime} does not follow the Edm.DateTimeOffset pattern: YYYY-MM-DDTHH:mm(:ss(.SSS))Z`
+    );
   }
-
-  return `/Date(${timestamp})/`;
+  return parsed;
 }
 
-/**
- * @hidden
- */
-export function parseNumber(value: string | number): number {
-  if (typeof value === 'number') {
-    return value;
+function edmDurationToMoment(value: string): Duration {
+  const durationPattern = /([\+,\-]{1,1})?P(\d+D)?T(\d{1,2}H)?(\d{1,2}M)?(\d{1,2}S)?(\d{2,2}\.\d+S)?/;
```

Comment:
```suggestion
  const durationPattern = /([\+,\-]{1,1})?P(\d{1,2})?(T(\d{1,2}H)?(\d{1,2}M)?(\d{1,2}S)?(\d{2,2}\.\d+S)?)?/;
```

Two points here:
* The D block can also have only two digits.
* The whole block including T is optional, if I read it correctly.

Reply:
Maybe the optional T block is irrelevant?

Reply:
I did not understand the spec placeholders so well. If you say it can only have two digits and the whole T block is irrelevant I adjust it and add also tests.